contracts/token/ERC20/ERC20.sol

```diff
@@ -224,7 +227,7 @@ contract ERC20 is Context, IERC20, IERC20Metadata {
         address to,
         uint256 amount
     ) internal virtual {
-        require(from != address(0) || to != address(0), "ERC20: invalid transfer operation");
+        require(!(from == address(0) && to == address(0)), "ERC20: invalid transfer operation");
 
         uint256 fromBalance = _balances[from];
```

Comment:
We are reading this from storage even during mints when it's not necessary. Please move this inside the branches where it needs to be read.